datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/dataset/DatasetType.java

```diff
@@ -53,7 +56,7 @@
 import static com.linkedin.metadata.Constants.*;
 
 
-public class DatasetType implements SearchableEntityType<Dataset>, BrowsableEntityType<Dataset>,
+public class DatasetType implements SearchableEntityType<Dataset, Pair<String, String>>, BrowsableEntityType<Dataset, Pair<String, String>>,
```

Comment:
Can we strongly type this instead of using Pair? We need to clearly state that this is urn and a version

Reply:
Unfortunately, the interface we use for our GraphQL endpoints accepts a single key (see DataLoader). I can do Pair<Urn,String> I think, but that makes things a bit less clean down the line due to Urn->String conversions and pushes the exception handling for Urn conversion into the GmsGraphQLEngine lambda. 

The order is clear on the actual endpoint itself because these come from params.

Reply:
I wasn't necessarily referring to Urn vs string but more having an actual object with two variables urn and version. Since Pair it is not clear what goes in the pair

Reply:
I used Pair to make the logic for the Rest.li client code easier and I just used it throughout. I think changing it around is doable.

Reply:
Yeah let's do that. I think should be pretty simple since Dataset is the only one that uses it at least for now?